protocol/x/ratelimit/keeper/keeper_test.go

```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
 }
 
 func TestSetGetLimitParams_Success(t *testing.T) {
-	tApp := testapp.NewTestAppBuilder(t).Build()
-	ctx := tApp.InitChain()
-	k := tApp.App.RatelimitKeeper
-
-	limiters := []types.Limiter{
-		{
-			PeriodSec:       3_600,
-			BaselineMinimum: dtypes.NewInt(100_000_000_000), // 100k tokens (assuming 6 decimals)
-			BaselineTvlPpm:  10_000,                         // 1%
+	// Run tests.
+	tests := map[string]struct {
+		denom                string
+		balances             []banktypes.Balance
+		limiters             []types.Limiter
+		expectedCapacityList []dtypes.SerializableInt
+	}{
+		"0 TVL, capactiy correctly initialized as minimum baseline": {
+			denom: testDenom,
+			limiters: []types.Limiter{
+				{
+					PeriodSec:       3_600,
+					BaselineMinimum: dtypes.NewInt(100_000_000_000), // 100k tokens (assuming 6 decimals)
+					BaselineTvlPpm:  10_000,                         // 1%
+				},
+				{
+					PeriodSec:       86_400,
+					BaselineMinimum: dtypes.NewInt(1_000_000_000_000), // 1m tokens (assuming 6 decimals)
+					BaselineTvlPpm:  100_000,                          // 10%
+				},
+			},
+			expectedCapacityList: []dtypes.SerializableInt{
+				dtypes.NewInt(100_000_000_000),
+				dtypes.NewInt(1_000_000_000_000),
+			},
 		},
-		{
-			PeriodSec:       86_400,
-			BaselineMinimum: dtypes.NewInt(1_000_000_000_000), // 1m tokens (assuming 6 decimals)
-			BaselineTvlPpm:  100_000,                          // 10%
+		"50m TVL, capactiy correctly initialized to 1% and 10%": {
+			denom: testDenom,
+			limiters: []types.Limiter{
+				{
+					PeriodSec:       3_600,
+					BaselineMinimum: dtypes.NewInt(100_000_000_000), // 100k tokens (assuming 6 decimals)
+					BaselineTvlPpm:  10_000,                         // 1%
+				},
+				{
+					PeriodSec:       86_400,
+					BaselineMinimum: dtypes.NewInt(1_000_000_000_000), // 1m tokens (assuming 6 decimals)
+					BaselineTvlPpm:  100_000,                          // 10%
+				},
+			},
+			balances: []banktypes.Balance{
```

Comment:
1. can we add a test case where there's some rounding down or rounding up happens?
2. corner cases: can we add test cases where baseline min and the ppm are 0 or maxmium? 

for example, 10M TVL and the base line ppm are 3% and 30% or something like that?

Reply:
1. Added unit test for rounding.